File: core/src/main/java/com/mcreater/amclcore/model/oauth/XSTSTokenResponseModel.java
package com.mcreater.amclcore.model.oauth;

import com.mcreater.amclcore.annotations.OAuthLoginModel;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
@OAuthLoginModel
public class XSTSTokenResponseModel {
    private XSTSTokenResponsePropertiesModel Properties;
    private String RelyingParty;
    private String TokenType;

    @Builder
    @Data
    @OAuthLoginModel
    public static class XSTSTokenResponsePropertiesModel {
        private String SandboxId;
        private List<String> UserTokens;
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/XSTSTokenResponseModel.kt
package com.mcreater.amclcore.model.oauth

import com.mcreater.amclcore.annotations.OAuthLoginModel

@OAuthLoginModel
data class XSTSTokenResponseModel(
    var Properties: XSTSTokenResponsePropertiesModel? = null,
    var RelyingParty: String? = null,
    var TokenType: String? = null
) {
    @OAuthLoginModel
    data class XSTSTokenResponsePropertiesModel(
        var SandboxId: String? = null,
        var UserTokens: List<String>? = null
    )
}
